ui/src/treeDataProvider/runWithLoadingNotification.ts

```diff
@@ -3,20 +3,31 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { CancellationToken, ProgressLocation, window } from 'vscode';
+import { CancellationToken, EventEmitter, ProgressLocation, window } from 'vscode';
 import * as types from '../../index';
 import { localize } from '../localize';
 
+let notificationActive: boolean = false;
+
 export async function runWithLoadingNotification<T>(context: types.ILoadingTreeContext, callback: (cancellationToken: CancellationToken) => Promise<T>): Promise<T> {
-    return await window.withProgress({ location: ProgressLocation.Notification, cancellable: true }, async (progress, cancellationToken) => {
-        const message: string = context.loadingMessage || localize('loadingAll', 'Loading resources...');
-        const messageDelay: number = context.loadingMessageDelay !== undefined ? context.loadingMessageDelay : 2;
-        const timer: NodeJS.Timer = setTimeout(() => progress.report({ message }), messageDelay * 1000);
+    if (notificationActive) {
+        return await callback({
+            isCancellationRequested: false,
+            onCancellationRequested: (new EventEmitter()).event
+        });
+    } else {
+        return await window.withProgress({ location: ProgressLocation.Notification, cancellable: true }, async (progress, cancellationToken) => {
+            notificationActive = true;
```

Comment:
In a pattern like this, there shouldn't be any code between `notificationActive = true;` and the start of the try block so that we're 100% certain it gets reset in the `finally` block.